src/collective/purgebyid/events.py

```diff
@@ -62,11 +58,7 @@ def mutate(self):
 
 @adapter(IPubAfterTraversal)
 def handle_request_after_traversal(event):
-    """handle "IPubAfterTraversal".
-
-    TODO: all the objects traversed are involved or only the last (the
-          firstest within request.PARENTS)?
-    """
+    """handle "IPubAfterTraversal"."""
```

Comment:
I have to double-check, but if I remember correctly, the first `parent` of the request is the published context itself. Am I wrong?

https://github.com/collective/collective.purgebyid/commit/b7610b8270aac6ac764fb606c53da02fe9ca2d13

Reply:
I inspected the results for a simple request (Plone/front-page). The first parent is not the same as the published object. However the published content is usually the view responsible for rendering (which contains no IUUID) and the first parent is typically the actual content we are interested in. So this just works fine.

As a side note when calling a simple file from a resource directory, that FilesystemFile is the published content and the FilesystemResourceDirectory is the first parent. I'm not sure if we need to handle those cases anyway, because files do not need to be purged? In the end, this is a topic out of scope for this PR.

tl;dr LGTM